src/SWP/MultiTenancyBundle/Resolver/TenantResolver.php

```diff
@@ -0,0 +1,101 @@
+<?php
+
+/**
+ * This file is part of the Superdesk Web Publisher MultiTenancy Bundle.
+ *
+ * Copyright 2015 Sourcefabric z.u. and contributors.
+ *
+ * For the full copyright and license information, please see the
+ * AUTHORS and LICENSE files distributed with this source code.
+ *
+ * @copyright 2015 Sourcefabric z.ú.
+ * @license http://www.superdesk.org/license
+ */
+namespace SWP\MultiTenancyBundle\Resolver;
+
+use Symfony\Component\HttpKernel\Exception\NotFoundHttpException;
+use Doctrine\ORM\EntityRepository;
+
+class TenantResolver implements TenantResolverInterface
+{
+    /**
+     * @var string
+     */
+    protected $domain;
+
+    /**
+     * @var EntityRepository
+     */
+    protected $tenantRepository;
+
+    /**
+     * Construct.
+     *
+     * @param string           $domain
+     * @param EntityRepository $tenantRepository
+     */
+    public function __construct($domain, EntityRepository $tenantRepository)
+    {
+        $this->domain = $domain;
+        $this->tenantRepository = $tenantRepository;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function resolve($host = null)
+    {
+        if ($host === $this->domain) {
+            throw new \Exception('Cannot resolve the root domain. Provide subdomain instead.');
+        }
+
+        $subdomain = $this->extractSubdomain($host);
+        $tenant = $this->tenantRepository->findOneBy(['subdomain' => $subdomain]);
+        if (!$tenant) {
+            throw $this->createNotFoundException(sprintf(
+                'Subdomain "%s" at "%s" host not found!',
+                $subdomain,
+                $this->domain
+            ));
+        }
+
+        return $tenant;
+    }
+
+    /**
+     * Extracts subdomain from the host.
+     *
+     * @param string $host Hostname
+     *
+     * @return string|null
+     */
+    public function extractSubdomain($host)
+    {
+        $parts = explode('.', str_replace('.'.$this->domain, '', $host));
+        $subdomain = null;
+        if (count($parts) === 1 && $parts[0] !== 'www') {
+            $subdomain = $parts[0];
+        }
+
+        return $subdomain;
+    }
+
+    /**
+     * Returns a NotFoundHttpException.
+     *
+     * This will result in a 404 response code. Usage example:
+     *
+     *     throw $this->createNotFoundException('Subdomain not found!');
+     *
+     * @param string          $message  A message
+     * @param \Exception|null $previous The previous exception
+     *
+     * @return NotFoundHttpException
+     *
+     * @see https://github.com/symfony/framework-bundle/blob/master/Controller/Controller.php#L253
+     */
+    protected function createNotFoundException($message = 'Not Found', \Exception $previous = null)
```

Comment:
if we use it once in this class then we don't need function for one line of code ;)